### ISSUE TITLE
Support Locality hierarchy

### ISSUE THREAD

**Issue by zednis** [tetherless-world/dtdi-mineral-observation-map#17]
The Localities seem to have a basic hierarchy that we are not taking into account.

It would be nice if the locality facet contained parent localities who were associated with observations made at all of their child localities.

This would enable users to start with the parent localities (which would have higher observation counts) and drill down into more specific localities.

It would also cause observations made at child localities to show up when the user uses the freetext search box with text that hits on the name of a parent locality.

We should be able to support this by having parent locality information in each observation JSON.  We could updated the locality object in the JSON to include a transitiveLocalities property which would be an array of all localities (direct, parent, parent of parent, etc).  The locality facet would then be updated to use transitiveLocalities property.

@kolans @xgmachina

**Comment by zhongh** [comment]
@zednis @xgmachina @kolans   I agree wholeheartedly that locality hierarchy is very very important piece of information that we don't want to miss. However, according to a very early email conversation with Josh and as stated by the rruff website, Mineral Evolution database is currently under development. The locality information, the parent-chile location information, and etc, still seem to be not very well groomed, sometimes inconsistent, on the website at this point of time. 

I suggest that this week we can fix other issues on which we already conclusion, then we can have another branch working on the "locality hierarchy" (or whatever it should be called) after we chat together with Josh next week?
